src/functionaltests/java/com/ericsson/ei/notifications/ttl/TestTTLSteps.java

```diff
@@ -0,0 +1,232 @@
+
+        @Autowired
+        private MongoDBHandler mongoDBHandler;
+
+        @Autowired
+        private InformSubscription informSubscription;
+
+        @Before("@TestNotificationRetries")
+        public void beforeScenario() {
+                setUpMockServer();
+        }
+
+        @After("@TestNotificationRetries")
+        public void afterScenario() throws IOException {
+            LOGGER.debug("Shutting down mock servers.");
+            mockServerClient.close();
+            clientAndServer.stop();
+        }
+
+        // START TEST SCENARIOS
+
+        @Given("^Missed notification is created in database with index \"([A-Za-z0-9_]+)\"$")
+        public void missed_notification_is_created_in_database(
+            String indexName) throws IOException {
+            LOGGER.debug("Starting scenario @TestTTL");
+            missedNotification = FileUtils.readFileToString(new File(MISSED_NOTIFICATION_FILE_PATH), "utf-8");
+            BasicDBObject missedNotificationDocument = prepareDocumentWithIndex(missedNotification,
+                    indexName);
+
+            // setting 1 second TTL on index in db
+            mongoDBHandler.createTTLIndex(missedNotificationDatabase, missedNotificationCollectionName,
+                    indexName, 1);
+            Boolean isInserted = mongoDBHandler
+                    .insertDocument(missedNotificationDatabase,
+                            missedNotificationCollectionName,
+                            missedNotificationDocument.toString());
+            assertEquals(true, isInserted);
+
+            List<String> result = mongoDBHandler.getAllDocuments(missedNotificationDatabase,
+                    missedNotificationCollectionName);
+            assertEquals("Database " + missedNotificationDatabase + " is not empty.", false, result.isEmpty());
+        }
+
+        @Given("^Aggregated object is created in database with index \"([A-Za-z0-9_]+)\"$")
+        public void aggregated_object_is_created_in_database(
+            String indexName) throws IOException {
+            aggregatedObject = FileUtils.readFileToString(new File(AGGREGATED_OBJECT_FILE_PATH), "utf-8");
+            BasicDBObject aggregatedDocument = prepareDocumentWithIndex(aggregatedObject, indexName);
+
+            // setting 1 second TTL on index in db
+            mongoDBHandler.createTTLIndex(aggregationDataBaseName, aggregationCollectionName,
+                    indexName, 1);
+            Boolean isInserted = mongoDBHandler.insertDocument(aggregationDataBaseName,
+                    aggregationCollectionName, aggregatedDocument.toString());
+            assertEquals(true, isInserted);
+
+            List<String> result = mongoDBHandler.getAllDocuments(aggregationDataBaseName,
+                    aggregationCollectionName);
+            assertEquals("Database " + aggregationDataBaseName + " is not empty.", false, result.isEmpty());
+        }
+
+        @When("^I sleep for \"([^\"]*)\" seconds")
```

Comment:
What if instead of sleeping for 60 seconds always, rewrite the check in "xx has been deleted from yy" to check for 60 seconds? That way if it was deleted after 30 seconds the test could continue more quickly.

So the When could be "{DB-entry} has been deleted from {database} within {seconds}"

Reply:
Sleeping is that a step step?  I think this should be moved to the end of the previoues test step.
A test step should test something and do some assertion on some actual values and compare it with some expected values.